### ISSUE TITLE
Sharing: Share by email form is not accessible for screenreaders.

### ISSUE THREAD

**Issue by danjjohnson** [Automattic/jetpack#18310]
When clicking the share by email button, the code for the form that displays on screen is actually rendered at the end of the page, so screenreaders don't notice that a form has been added. This makes the feature appear not to work to people using a screenreader.

#### Describe the solution you'd like
When you click the button, there should be some way for screenreaders to know that a form is now available.

#### Additional context

Reported here: https://wordpress.org/support/topic/accessibility-of-posts-publicized-to-facebook/

**Comment by joedolson** [comment]
Additional issues with the share by email form:

- The pop-up that displays the email form does not receive focus when it opens, so there's no easy way for a user to get to the form. - The form source is located deep in the footer, so if they do find the form, they're not in a location that's in any proximity to the content they're trying to reference. 
- The link to the email form does not provide any stateful information (aria-expanded attribute) to indicate whether or not the form is currently open. The link should operate as a disclosure widget (preferably using a button), communicating the current status of the control. Modal toggles don't need to indicate current state, but a modal wouldn't be a good choice for this UI, since the user needs access to the content they're trying to share.

**Comment by jeherve** [comment]
This can now be closed since we now use mailto instead of a custom email sharing button (see #24040).
